diff -ruN elfio-orig/elfio_dump.hpp elfio/elfio_dump.hpp
--- elfio-orig/elfio_dump.hpp	2013-01-30 15:26:56.000000000 -0500
+++ elfio/elfio_dump.hpp	2013-08-09 14:01:56.208746796 -0400
@@ -27,7 +27,7 @@
 #include <ostream>
 #include <sstream>
 #include <iomanip>
-#include <elfio/elfio.hpp>
+#include "elfio.hpp"
 
 namespace ELFIO {
 
diff -ruN elfio-orig/elfio.hpp elfio/elfio.hpp
--- elfio-orig/elfio.hpp	2013-05-16 11:37:39.000000000 -0400
+++ elfio/elfio.hpp	2013-08-09 14:01:56.208746796 -0400
@@ -32,16 +32,16 @@
 
 #include <string>
 #include <fstream>
-#include <algorithm>
+//#include <algorithm>
 #include <vector>
 #include <typeinfo>
 
-#include <elfio/elf_types.hpp>
-#include <elfio/elfio_utils.hpp>
-#include <elfio/elfio_header.hpp>
-#include <elfio/elfio_section.hpp>
-#include <elfio/elfio_segment.hpp>
-#include <elfio/elfio_strings.hpp>
+#include "elf_types.hpp"
+#include "elfio_utils.hpp"
+#include "elfio_header.hpp"
+#include "elfio_section.hpp"
+#include "elfio_segment.hpp"
+#include "elfio_strings.hpp"
 
 #define ELFIO_HEADER_ACCESS_GET( TYPE, FNAME ) \
 TYPE                                           \
@@ -619,10 +619,10 @@
 
 } // namespace ELFIO
 
-#include <elfio/elfio_symbols.hpp>
-#include <elfio/elfio_note.hpp>
-#include <elfio/elfio_relocation.hpp>
-#include <elfio/elfio_dynamic.hpp>
+#include "elfio_symbols.hpp"
+#include "elfio_note.hpp"
+#include "elfio_relocation.hpp"
+#include "elfio_dynamic.hpp"
 
 #ifdef _MSC_VER
 #pragma warning ( pop )
diff -ruN elfio-orig/elfio_segment.hpp elfio/elfio_segment.hpp
--- elfio-orig/elfio_segment.hpp	2013-05-01 15:17:01.000000000 -0400
+++ elfio/elfio_segment.hpp	2013-08-09 14:06:19.509426964 -0400
@@ -43,6 +43,7 @@
     ELFIO_GET_SET_ACCESS_DECL( Elf64_Addr, physical_address );
     ELFIO_GET_SET_ACCESS_DECL( Elf_Xword,  file_size        );
     ELFIO_GET_SET_ACCESS_DECL( Elf_Xword,  memory_size      );
+    ELFIO_GET_ACCESS_DECL    ( Elf64_Off,  file_offset      );
 
     virtual const char* get_data() const = 0;
 
@@ -86,6 +87,7 @@
     ELFIO_GET_SET_ACCESS( Elf64_Addr, physical_address, ph.p_paddr  );
     ELFIO_GET_SET_ACCESS( Elf_Xword,  file_size,        ph.p_filesz );
     ELFIO_GET_SET_ACCESS( Elf_Xword,  memory_size,      ph.p_memsz  );
+    ELFIO_GET_ACCESS    ( Elf64_Off,  file_offset,      ph.p_offset );
 
 //------------------------------------------------------------------------------
     Elf_Half
